com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/CheckForOperationsInProgressStep.java

```diff
@@ -0,0 +1,150 @@
+package com.sap.cloud.lm.sl.cf.process.steps;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.apache.commons.collections4.MapUtils;
+import org.cloudfoundry.client.lib.CloudControllerClient;
+import org.cloudfoundry.client.lib.domain.CloudService;
+import org.springframework.stereotype.Component;
+
+import com.sap.cloud.lm.sl.cf.client.lib.domain.CloudServiceExtended;
+import com.sap.cloud.lm.sl.cf.client.lib.domain.ImmutableCloudServiceExtended;
+import com.sap.cloud.lm.sl.cf.core.cf.clients.EventsGetter;
+import com.sap.cloud.lm.sl.cf.core.cf.clients.ServiceGetter;
+import com.sap.cloud.lm.sl.cf.core.cf.services.ServiceOperation;
+import com.sap.cloud.lm.sl.cf.core.cf.services.ServiceOperationState;
+import com.sap.cloud.lm.sl.cf.core.cf.services.ServiceOperationType;
+import com.sap.cloud.lm.sl.cf.process.message.Messages;
+import com.sap.cloud.lm.sl.common.util.JsonUtil;
+
+@Component("checkForOperationsInProgressStep")
+public class CheckForOperationsInProgressStep extends AsyncFlowableStep {
+
+    private static final String LAST_SERVICE_OPERATION = "last_operation";
+    private static final String SERVICE_OPERATION_TYPE = "type";
+    private static final String SERVICE_OPERATION_STATE = "state";
+    private static final String SERVICE_OPERATION_DESCRIPTION = "description";
+
+    @Inject
+    private ServiceGetter serviceInstanceGetter;
+    @Inject
+    private EventsGetter eventsGetter;
+
+    @Override
+    protected StepPhase executeAsyncStep(ExecutionWrapper execution) throws Exception {
+        CloudControllerClient controllerClient = execution.getControllerClient();
+        String spaceId = StepsUtil.getSpaceId(execution.getContext());
+        List<CloudServiceExtended> servicesToProcess = getServices(execution);
+        List<CloudServiceExtended> existingServices = getExistingServices(servicesToProcess, controllerClient);
+
+        if (existingServices.isEmpty()) {
+            return StepPhase.DONE;
+        }
+        Map<CloudServiceExtended, ServiceOperation> servicesInProgressState = getServicesOperationInProgress(existingServices,
```

Comment:
The variable and the method names are different. Either refactor the name of the variable or change the name of the method

Reply:
done